src/PointwiseFunctions/Elasticity/PotentialEnergy.hpp

```diff
@@ -0,0 +1,69 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Tags.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/ConstitutiveRelation.hpp"
+#include "PointwiseFunctions/Elasticity/ConstitutiveRelations/Tags.hpp"
+
+namespace Elasticity {
+
+// @{
+/*!
+ * \brief The potential energy density \f$U=-\frac{1}{2}S_{ij}T^{ij}\f$ stored
+ * in the deformation of the elastic material (see Eq. 11.25 in
+ * \cite ThorneBlandford2017)
+ *
+ * Note that the two-dimensional instantiation of this function assumes that
+ * only the terms of \f$S_{ij}T^{ij}\f$ where both \f$i\f$ and \f$j\f$
+ * correspond to one of the computational dimensions contribute to the sum. This
+ * is the case for the plane-stress approximation employed in the
+ * two-dimensional `Elasticity::ConstitutiveRelations::IsotropicHomogeneous`,
+ * for example, where \f$T^{i3}=0=T^{3i}\f$.
+ */
+template <size_t Dim>
+void potential_energy_density(
+    gsl::not_null<Scalar<DataVector>*> potential_energy_density,
+    const tnsr::ii<DataVector, Dim>& strain,
+    const tnsr::I<DataVector, Dim>& coordinates,
+    const ConstitutiveRelations::ConstitutiveRelation<Dim>&
+        constitutive_relation) noexcept;
+
+template <size_t Dim>
+Scalar<DataVector> potential_energy_density(
+    const tnsr::ii<DataVector, Dim>& strain,
+    const tnsr::I<DataVector, Dim>& coordinates,
+    const ConstitutiveRelations::ConstitutiveRelation<Dim>&
+        constitutive_relation) noexcept;
+// @}
+
+namespace Tags {
+
+/// \brief Computes the energy density stored in the deformation of the elastic
+/// material.
+/// \see Elasticity::potential_energy_density
+template <size_t Dim>
+struct PotentialEnergyCompute : Elasticity::Tags::PotentialEnergy<Dim>,
```

Comment:
Why do we need a compute tag, instead of just calling the function? I don't know the elliptic code's details, but in general, we'd like to avoid compute tags unless necessary

Reply:
We want to observe the potential energy density and its integral over the computational domain, if I understand correctly compute tags are still the best way to do that.